### PR TITLE
Signup flow front

### PR DESCRIPTION
**Pull request by SamerElaila**
Add form for teachers and hosts, and add upload image to AWS S3 functionality

### CODE REVIEW

File: client/src/components/stripe/LoadingStripe.js [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
+    if (isNewUser) {
+      return <Redirect to='/users/new' />;
+    } else {
+      if (isTeacher) {
+        return <Redirect to='/dashboard/teachers' />;
+      } else if (isHost) {
+        return <Redirect to='/dashboard/hosts' />;
+      } else {
+        return (
+          <div className='loading-stripe__wrapper'>
```
Comment (des-des) [inline client/src/components/stripe/LoadingStripe.js]:
```
if (isNewUser) {
  ...
} else {
    if (isTeacher) }{
        ...
    } else if (isHost) {
        ...
    } else  {
      // Please dont do this ...
    }

File: client/src/components/stripe/LoadingStripe.js [hunk 2]
```diff
@@ -1,33 +1,44 @@
...
+            <PulseLoader color={'#123abc'} loading={true} />
+          </div>
```
Comment (des-des) [inline client/src/components/stripe/LoadingStripe.js]: Okay, these conditions are weird
```js
if (!newUser && !teacher && !host) {
    isLoading
}
```
Can you
  1. Un-nest these if statements
  2. See if there is a better bool in state to represent isLoading

File: client/src/sagas/signupForm.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { call, put } from 'redux-saga/effects';
+
+// eslint-disable-next-line no-unused-vars
```
Comment (des-des) [inline client/src/sagas/signupForm.js]: I think we do not need this

File: client/src/sagas/signupForm.js [hunk 4]
```diff
@@ -17,13 +16,20 @@ const signupFormFailed = () => ({ type: SIGN_UP_FORM_FAILED });
...
+      API.signS3Url,
+      data.avatar.name,
+      data.avatar.type
+    );
+    const uploadResponse = yield call(
+      API.uploadToS3,
+      signUrlResponse.data.signedUrl,
+      data.avatar
+    );
+    yield put(signupFormSuccess(uploadResponse.data.imageUrl));
```
Comment (des-des) [inline client/src/sagas/signupForm.js]: I feel like we are missing the 3rd request.
  1. GetSignedUrl
  2. Upload
  3. Submit form
?

File: client/src/components/signupForm/index.js [hunk 5]
```diff
@@ -13,49 +13,43 @@ class SignupForm extends Component {
...
+      { name: 'inviteCode', label: 'Invite Code' }
+    ];
+
     this.handleSubmit = this.handleSubmit.bind(this);
     this.handleChange = this.handleChange.bind(this);
   }
 
   handleSubmit(event) {
     event.preventDefault();
+    console.log(this.state);
```
Comment (des-des) [inline client/src/components/signupForm/index.js]: :fire:

File: client/src/sagas/signupForm.js (hunk 4 above)
Comment (SamerElaila) [reply client/src/sagas/signupForm.js]: Oops I missed it!

**Pull request merged**